xonsh/history.py

```diff
@@ -407,103 +406,84 @@ def _hist_create_parser():
     return p
 
 
-def _hist_show(ns=None, hist=None, start_index=None, end_index=None,
-               start_time=None, end_time=None, location=None):
-    """Show the requested portion of shell history.
-    Accepts multiple history sources (xonsh, bash, zsh)
-
-    May be invoked as an alias with history all/bash/zsh which will
-    provide history as stdout or with __xonsh_history__.show()
-    which will return the history as a list with each item
-    in the tuple form (name, start_time, index).
+def _hist_get_session(session='session', location=None):
+    "Yield commands from history."
+    cmds = _HIST_SESSIONS[session](location=location)
+    try:
+        yield from cmds
+    except TypeError:
+        pass
+
+
+def _hist_get_portion(commands, slices):
+    "Yield from portions of history commands. "
+    commands = list(commands)
+    for s in slices:
+        s = ensure_slice(s)
+        yield from commands[s]
+
+
+def _hist_filter_ts(commands, start_time=None, end_time=None):
+    """Yield only the commands between start and end time."""
+    if start_time is None:
+        start_time = 0.0
+    if end_time is None:
+        end_time = float('inf')
+    for cmd in commands:
+        if end_time > cmd[1] > start_time:
```

Comment:
Well, probably not both.  The Pythonic thing is to be inclusive on the lower end and exclusive on the upper end.